app/services/gogox.rb

```diff
@@ -0,0 +1,129 @@
+  #     "contact" => {
+  #       "name" => "Swati J",
+  #       "phone_number" => "+85251111113",
+  #       "phone_extension" => nil
+  #     }
+  #   },
+  #   "destinations" => [{
+  #     "name" => "GCAdmin User",
+  #     "street_address" => "Santa Peak Road",
+  #     "floor_or_unit_number" => nil,
+  #     "location" => {"lat" => 32.3700365, "lng" => 120.9930016},
+  #     "contact" => {
+  #       "name" => "GCAdmin User",
+  #       "phone_number" => "+85251111111"
+  #     }
+  #   }],
+  #   "note_to_courier" => nil,
+  #   "price" => {"amount" => 15000, "currency" => "HKD"},
+  #   "price_breakdown" => [{"key" => "fee", "amount" => 15000}]
+  # }
+  def book
+    GogoxApi::Transport.new(order_attributes).order
+  end
+
+  # Response:
+  # {
+  #   "vehicle_type" => "van",
+  #   "estimated_price" => {"amount" => 15000, "currency" => "HKD"},
+  #   "estimated_price_breakdown" => [{"key" => "fee", "amount" => 15000}]
+  # }
+  def quotation
+    GogoxApi::Transport.new(quotation_attributes).quotation
+  end
+
+  class << self
+
+    def transport_status(booking_id)
+      GogoxApi::Transport.new.status(booking_id)
+    end
+
+    # Response
+    # Response is nil on successful cancellation of GOGOX transport
+    def cancel_order(booking_id)
+      response = GogoxApi::Transport.new.cancel(booking_id)
+      if !response
+        {
+          order_uuid: booking_id,
+          status: "cancelled"
+        }
+      end
+    end
+
+  end
+
+  private
+
+  def order_attributes
+    {
+      'vehicle_type': vehicle_type,
+      "pickup_location": params[:pickup_location],
+      "pickup_street_address": params[:pickup_street_address],
+      "scheduled_at": parse_time,
+      "pickup_contact_name": params[:pickup_contact_name],
+      "pickup_contact_phone": params[:pickup_contact_phone],
+      "destination_location": params[:destination_location],
+      "destination_street_address": params[:destination_street_address],
+      "destination_contact_name": params[:destination_contact_name],
+      "destination_contact_phone": params[:destination_contact_phone]
+    }
+  end
+
+  def quotation_attributes
+    {
+      'vehicle_type': vehicle_type,
+      "scheduled_at": parse_time,
+      "pickup_location": params[:pickup_location],
+      "destination_location": params[:destination_location]
+    }
+  end
+
+  def vehicle_type
+    if vehicle.blank? || !VEHICLE_TYPES.include?(vehicle)
+      raise(ValueError, "vehicle should be from #{VEHICLE_TYPES.join(', ')}")
+    end
+    vehicle
+  end
+
+  def parse_pickup_time(time = nil)
+    return time if time.present?
+
+    # next available date within next 5 days
+    next_available_date = DateSet.new(5, 1).available_dates.first
+    (next_available_date.beginning_of_day + 12.hours)
+  end
+
+  def parse_time
+    @time = DateTime.parse(@time.to_s)
+    @time = @time.zone == "HKT" ? @time : @time.in_time_zone("Asia/Hong_Kong")
+    @time.to_i
+  end
+
+  class ValueError < StandardError; end
```

Comment:
Instead of just value error defining here, create a error in errors.rb
something like TransportationServiceError